addons/im_livechat/models/mail_message_reaction.py

```diff
@@ -0,0 +1,21 @@
+from odoo import models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class MailMessageReaction(models.Model):
+    _name = "mail.message.reaction"
+    _inherit = ["mail.message.reaction"]
+
+    def _reacting_persona_to_store(self, store: Store):
+        channel_reactions_with_partner = self.filtered(
+            lambda r: r.partner_id and r.message_id.model == "discuss.channel"
+        )
+        channel_by_message = channel_reactions_with_partner.message_id._record_by_message()
+        livechat_agent_reactions = channel_reactions_with_partner.filtered(
+            lambda r: channel_by_message.get(r.message_id, self.env["discuss.channel"]).channel_type
+            == "livechat"
+        )
+        super(MailMessageReaction, self - livechat_agent_reactions)._reacting_persona_to_store(
+            store
+        )
+        store.add(livechat_agent_reactions.partner_id, fields=["avatar_128", "user_livechat_username"])
```

Comment:
This should as much as possible be common code with what we do in `_author_to_store` of message.

Reply:
It's very similar but I don't think there is much to put in common: 
- initial filter clearly depends on the model we're filtering so cannot be generic.
- second filter also depends on the model's field holding the message
- fields are different for the `store.add`
I wouldn't make a generic method for now.
We could probably simplify this by adding a `_partner_to_store(partner)` method on `discuss.channel` but then we would have to fetch the channels systematically while it's only done when live chat is installed for now.